tools/flaky-test-retryer/log_parser_test.go

```diff
@@ -0,0 +1,159 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"reflect"
+	"testing"
+
+	"knative.dev/test-infra/tools/flaky-test-reporter/jsonreport/fakejsonreport"
+	"knative.dev/test-infra/tools/monitoring/prowapi"
+)
+
+var (
+	fakeFlakyTests = []string{"test0", "test1", "test2"}
+	fakeWrongState = &prowapi.ReportMessage{
+		JobName: "fakejob",
+		JobType: prowapi.PresubmitJob,
+		Status:  prowapi.SuccessState,
+		Refs: []prowapi.Refs{{
+			Org:  "fakeorg",
+			Repo: fakeRepo,
+			Pulls: []prowapi.Pull{{
+				Number: 111,
+			}},
+		}},
+	}
+	fakeWrongType = &prowapi.ReportMessage{
+		JobName: "fakejob",
+		JobType: prowapi.PeriodicJob,
+		Status:  prowapi.FailureState,
+		Refs: []prowapi.Refs{{
+			Org:  "fakeorg",
+			Repo: fakeRepo,
+			Pulls: []prowapi.Pull{{
+				Number: 111,
+			}},
+		}},
+	}
+	fakeNoRefs = &prowapi.ReportMessage{
+		JobName: "fakejob",
+		JobType: prowapi.PresubmitJob,
+		Status:  prowapi.FailureState,
+		Refs:    nil,
+	}
+	fakeInvalidRepo = &prowapi.ReportMessage{
+		JobName: "fakejob",
+		JobType: prowapi.PresubmitJob,
+		Status:  prowapi.FailureState,
+		Refs: []prowapi.Refs{{
+			Org:  "fakeorg",
+			Repo: "invalid",
+			Pulls: []prowapi.Pull{{
+				Number: 111,
+			}},
+		}},
+	}
+	fakeNoPulls = &prowapi.ReportMessage{
+		JobName: "fakejob",
+		JobType: prowapi.PresubmitJob,
+		Status:  prowapi.FailureState,
+		Refs: []prowapi.Refs{{
+			Org:   "fakeorg",
+			Repo:  fakeRepo,
+			Pulls: nil,
+		}},
+	}
+	fakeValidMessage = &prowapi.ReportMessage{
+		JobName: "fakejob",
+		JobType: prowapi.PresubmitJob,
+		Status:  prowapi.FailureState,
+		Refs: []prowapi.Refs{{
+			Org:  "fakeorg",
+			Repo: fakeRepo,
+			Pulls: []prowapi.Pull{{
+				Number: 111,
+			}},
+		}},
+	}
+)
+
+func setup() {
+	client, _ = fakejsonreport.Initialize("")
+	client.CreateReport(fakeRepo, fakeFlakyTests, true)
+}
+
+func testGetFlakyTests(t *testing.T) {
+	data := []struct {
+		job       *JobData
+		wantArray []string
+		wantErr   error
+	}{
+		{&JobData{fakeValidMessage, nil, nil}, []string{"test0", "test1", "test2"}, nil},
+		{&JobData{fakeInvalidRepo, nil, nil}, []string{}, nil},
+	}
+	setup()
+	for _, test := range data {
+		gotArray, gotErr := test.job.getFlakyTests()
+		if !reflect.DeepEqual(gotArray, test.wantArray) {
+			t.Fatalf("Get Flaky Tests: got array %v, want array %v", gotArray, test.wantArray)
+		}
+		if gotErr != test.wantErr {
+			t.Fatalf("Get Flaky Tests: got err %v, want err %v", gotErr, test.wantErr)
+		}
+		if test.job.flakyReports == nil {
+			t.Fatalf("Get Flaky Tests: did not populate job cache")
+		}
+	}
+}
+
+func testIsSupported(t *testing.T) {
+	cases := []struct {
+		job  *JobData
+		want bool
+	}{
+		{&JobData{fakeWrongState, nil, nil}, false},
```

Comment:
I think these states should be moved in this function

Reply:
Add add a supported case

Reply:
moved as in reordered? also added that case.

Reply:
These vars (i.e. fakeWrongStaet) are not used outside of this test, should reside here

Reply:
done